hibernate-reactive-core/src/main/java/org/hibernate/reactive/event/impl/AbstractReactiveFlushingEventListener.java

```diff
@@ -129,9 +128,10 @@ private ReactiveActionQueue actionQueue(EventSource session) {
 		IdentitySet copiedAlready = new IdentitySet( 10 );
 		//safe from concurrent modification because of how concurrentEntries() is implemented on IdentityMap
 		return CompletionStages.loop(
-				Arrays.stream( persistenceContext.reentrantSafeEntityEntries() )
-						.filter( entry -> flushable( entry.getValue() ) ),
-				entry -> cascadeOnFlush( session, entry.getValue().getPersister(), entry.getKey(), copiedAlready )
+				persistenceContext.reentrantSafeEntityEntries(),
+				entry -> flushable( entry.getValue() )
+						? cascadeOnFlush( session, entry.getValue().getPersister(), entry.getKey(), copiedAlready )
+						: voidFuture()
```

Comment:
I don't think this pattern will cause any issue but peraphs I could filter the array before doing the loop.

Reply:
Why don't you think it's an issue? It looks like a potentially serious degredation to me (but I might be wrong).